src/components/UserSearch.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import {Route} from 'react-router-dom';
+
+class UserSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      githubUsername: '',
+      selectedUsername: '',
+    };
+
+    this.searchUser = this.searchUser.bind(this);
+    this.updateUsername = this.updateUsername.bind(this);
+  }
+
+  searchUser(evt) {
+    evt.preventDefault();
+    // TODO: Dispatch action that will request data from githubs API
+    console.log("Searching user....");
+    this.setState({selectedUsername: this.state.githubUsername});
```

Comment:
The component's API for this could be something like `onUserSearch` with it taking in the current username as a parameter. A function callback that dispatches some action to request users matching the provided username would likely be injected in as the `onUserSearch` prop.